lib/public/components/Filters/RunsFilter/eorReasonsFilterModel.js

```diff
@@ -0,0 +1,119 @@
+ */
+export class EorReasonFilterModel extends Observable {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+        this._eorReasonTypes = RemoteData.notAsked();
+        this._fetchReasonTypes();
+        this._newEorReason = { category: emptyEorReason.category, title: emptyEorReason.title, description: emptyEorReason.description };
+    }
+
+    /**
+     * Retrieve a list of reason types from the API
+     *
+     * @returns {Promise<void>} resolves once the data has been fetched
+     */
+    async _fetchReasonTypes() {
+        this._eorReasonTypes = RemoteData.loading();
+        this.notify();
+
+        try {
+            const { data: reasonTypes } = await getRemoteData('/api/runs/reasonTypes');
+            this._eorReasonTypes = RemoteData.success(reasonTypes);
+        } catch (error) {
+            this._eorReasonTypes = RemoteData.failure(error);
+        }
+
+        this.notify();
+    }
+
+    /**
+     * Returns a boolean indicating whether this filter is considered empty
+     * @returns {boolean} indicating whether this filter is considered empty
+     */
+    isEmpty() {
+        return this._newEorReason.category == '' && this._newEorReason.title == '' && this._newEorReason.description == '';
+    }
+
+    /**
+     * Empties the filter. this.isEmpty() evaluates to true.
+     * @returns {void}
+     */
+    reset() {
+        this._newEorReason = { category: emptyEorReason.category, title: emptyEorReason.title, description: emptyEorReason.description };
+    }
+
+    /**
+     * Gets an object of url params to value for the api query
+     * @returns {Object} url query parameter to parameter value
+     */
+    getFilterQueryParams() {
+        return {
+            ...!this.newEorReason.category == '' && { 'filter[eor][category]': this.newEorReason.category },
+            ...!this.newEorReason.title == '' && { 'filter[eor][title]': this.newEorReason.title },
+            ...!this.newEorReason.description == '' && { 'filter[eor][description]': this.newEorReason.description },
+        };
+    }
+
+    /**
+     * Getter for the EOR reason type
+     *
+     * @return {RemoteData} the EOR reason type
+     */
+    get eorReasonTypes() {
+        return this._eorReasonTypes;
+    }
+
+    /**
+     * Getter for the EOR reason types
+     * @param {Object} reasonTypes object representing the possible EOR reason types
+     * @return {void}
+     */
+    set eorReasonTypes(reasonTypes) {
+        this._eorReasonTypes = reasonTypes;
+    }
+
+    /**
+     * Getter for the EOR reason type
+     *
+     * @return {RemoteData} the EOR reason type
+     */
+    get newEorReason() {
+        return this._newEorReason;
+    }
+
+    /**
+     * Setter for the EOR reason type
+     * @param {Object} reason the object representing the selected EOR reason
+     * @return {void}
+     */
+    set newEorReason(reason) {
+        this._newEorReason = reason;
```

Comment:
The setter should inlcude the `notify` call as well
Moreover, it should apply default values for `category, title, description` if value is missing

Reply:
Should the call to `notify` in the view be removed in this case? such as [here](https://github.com/AliceO2Group/Bookkeeping/pull/1083/files/8075b4d7f34e60fd8c7e70707d13a17f19b2a2de#diff-10856bbc1f2c8b70f0214661ce0c66ae15e9220eee27aba28e2f6b474877244cR37)

Reply:
Yes I think, you mentioned this below!